docs/src/main/sphinx/security/internal-communication.rst

```diff
@@ -25,107 +26,32 @@ command:
 Internal SSL/TLS configuration
 ------------------------------
 
-SSL/TLS is configured in the ``config.properties`` file.  The SSL/TLS on the
-worker and coordinator nodes are configured using the same set of properties.
-Every node in the cluster must be configured. Nodes that have not been
-configured, or are configured incorrectly, are not able to communicate with
-other nodes in the cluster.
+The SSL/TLS on the worker and coordinator nodes are configured using the same
+set of properties. Every node in the cluster must be configured. Nodes that have
+not been configured, or are configured incorrectly, are not able to communicate
+with other nodes in the cluster.
 
-To enable SSL/TLS for Trino internal communication, do the following:
+In typical deployments you want to enable :doc:`/security/tls` for access to the
+cluster, before enabling SSL/TLS for Trino internal communication between the
+coordinator and all workers.
 
-1. Disable HTTP endpoint.
+Ensure the following configuration is identical on all cluster nodes.
 
-   .. code-block:: text
+1. Configure a shared secret for internal communication as described in
+   the preceding section.
 
-       http-server.http.enabled=false
+2. Enable automatic certificate creation and trust setup in
+   ``etc/config.properties``:
 
-   .. warning::
+   .. code-block:: properties
 
-       You can enable HTTPS, while leaving HTTP enabled. In most cases this is a
-       security hole. If you are certain you want to use this configuration, you
-       should consider using an firewall to limit access to the HTTP endpoint to
-       only those hosts that should be allowed to use it.
+     http-server.https.automatic-shared-secret=automatic-secret
```

Comment:
This is wrong. The property is internal and will be set automatically when you enable secure internal communication.

Reply:
I will check with Dain .. I am not sure what is really needed now.